src/main/java/appeng/items/storage/ItemStorageCellBase.java

```diff
@@ -0,0 +1,271 @@
+	{
+		final IMEInventoryHandler<?> inventory = AEApi.instance().registries().cell().getCellInventory( stack, null, getChannel() );
+
+		if( inventory instanceof ICellInventoryHandler )
+		{
+			final ICellInventoryHandler handler = (ICellInventoryHandler) inventory;
+			final ICellInventory cellInventory = handler.getCellInv();
+
+			if( cellInventory != null )
+			{
+				lines.add( cellInventory.getUsedBytes() + " " + GuiText.Of.getLocal() + ' ' + cellInventory.getTotalBytes() + ' ' + GuiText.BytesUsed.getLocal() );
+
+				lines.add( cellInventory.getStoredItemTypes() + " " + GuiText.Of.getLocal() + ' ' + cellInventory.getTotalItemTypes() + ' ' + GuiText.Types.getLocal() );
+
+				if( handler.isPreformatted() )
+				{
+					final String list = ( handler.getIncludeExcludeMode() == IncludeExclude.WHITELIST ? GuiText.Included : GuiText.Excluded ).getLocal();
+
+					if( handler.isFuzzy() )
+					{
+						lines.add( GuiText.Partitioned.getLocal() + " - " + list + ' ' + GuiText.Fuzzy.getLocal() );
+					}
+					else
+					{
+						lines.add( GuiText.Partitioned.getLocal() + " - " + list + ' ' + GuiText.Precise.getLocal() );
+					}
+				}
+			}
+		}
+	}
+
+	@Override
+	public int getBytes( final ItemStack cellItem )
+	{
+		return this.totalBytes;
+	}
+
+	@Override
+	public int getTotalTypes( final ItemStack cellItem )
+	{
+		return 63;
+	}
+
+	@Override
+	public boolean isBlackListed( final ItemStack cellItem, final T requestedAddition )
+	{
+		return false;
+	}
+
+	@Override
+	public boolean storableInStorageCell()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isStorageCell( final ItemStack i )
+	{
+		return true;
+	}
+
+	@Override
+	public String getUnlocalizedGroupName( final Set<ItemStack> others, final ItemStack is )
+	{
+		return GuiText.StorageCells.getUnlocalized();
+	}
+
+	@Override
+	public boolean isEditable( final ItemStack is )
+	{
+		return true;
+	}
+
+	@Override
+	public IItemHandler getUpgradesInventory( final ItemStack is )
+	{
+		return new CellUpgrades( is, 2 );
+	}
+
+	@Override
+	public IItemHandler getConfigInventory( final ItemStack is )
+	{
+		return new CellConfig( is );
+	}
+
+	@Override
+	public FuzzyMode getFuzzyMode( final ItemStack is )
+	{
+		final String fz = Platform.openNbtData( is ).getString( "FuzzyMode" );
+		try
+		{
+			return FuzzyMode.valueOf( fz );
+		}
+		catch( final Throwable t )
+		{
+			return FuzzyMode.IGNORE_ALL;
+		}
+	}
+
+	@Override
+	public void setFuzzyMode( final ItemStack is, final FuzzyMode fzMode )
+	{
+		Platform.openNbtData( is ).setString( "FuzzyMode", fzMode.name() );
+	}
+
+	@Override
+	public ActionResult<ItemStack> onItemRightClick( final World world, final EntityPlayer player, final EnumHand hand )
+	{
+		this.disassembleDrive( player.getHeldItem( hand ), world, player );
+		return new ActionResult<>( EnumActionResult.SUCCESS, player.getHeldItem( hand ) );
+	}
+
+	private boolean disassembleDrive( final ItemStack stack, final World world, final EntityPlayer player )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "disassembleDrive" is 12 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "world". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)